wgsl/index.bs

```diff
@@ -3198,58 +3479,136 @@ the test name.
 * v-0028: A fallthrough statement must not appear as the last statement in last clause of a switch.
 
 
-# Built-in variables TODO # {#builtin-variables}
+# Built-in variables # {#builtin-variables}
 
-<div class='example' heading="Valid Built-in Decoration Identifiers">
-  <xmp>
-    [[builtin(position)]]
-          OpDecorate %gl_Position BuiltIn Position
+See [[#builtin-inputs-outputs]] for how to declare a built-in variable.
 
-    [[builtin(vertex_idx)]]
-          OpDecorate %gl_VertexIdx BuiltIn VertexIndex
+<table class='data'>
+  <thead>
+    <tr><td>Built-in<td>Stage<td>Input or Output<td>Store type<td>Description
+  </thead>
 
-    [[builtin(instance_idx)]]
-          OpDecorate %gl_InstanceId BuiltIn InstanceIndex
+  <tr><td>`vertex_idx`
+      <td>vertex
+      <td>in
+      <td>u32
+      <td width="50%">Index of the current vertex within the current API-level draw command,
+         independent of draw instancing.
+
+         For a non-indexed draw, the first vertex has an index equal to the `firstIndex` argument
+         of the draw, whether provided directly or indirectly.
+         The index is incremented by one for each additional vertex in the draw instance.
+
+         For an indexed draw, the index is equal to the index buffer entry for
+         vertex, plus the `baseVertex` argument of the draw, whether provided directly or indirectly.
+
+  <tr><td>`instance_idx`
+      <td>vertex
+      <td>in
+      <td>u32
+      <td width="50%">Instance index of the current vertex within the current API-level draw command.
+
+         The first instance has an index equal to the `firstIstance` argument of the draw,
+         whether provided directly or indirectly.
+         The index is incremented by one for each additional instance in the draw.
+
+  <tr><td>`position`
+      <td>vertex
+      <td>out
+      <td>vec4&lt;f32&gt;
+      <td width="50%">Position of the current vertex.
+
+  <tr><td>`frag_coord`
+      <td>fragment
+      <td>in
+      <td>vec4&lt;f32&gt;
+      <td width="50%">Coordinate of the current fragment.
+
+  <tr><td>`front_facing`
+      <td>fragment
+      <td>in
+      <td>u32
+      <td width="50%">Non-zero when the current fragment is on a front-facing primitive.
```

Comment:
this should probably link to https://gpuweb.github.io/gpuweb/#dom-gpurasterizationstatedescriptor-frontface